fineract-provider/src/main/java/org/apache/fineract/infrastructure/creditbureau/service/ThitsaWorksCreditBureauIntegrationWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,420 @@
+
+    @Autowired
+    public ThitsaWorksCreditBureauIntegrationWritePlatformServiceImpl(final PlatformSecurityContext context,
+            final FromJsonHelper fromApiJsonHelper, final TokenRepositoryWrapper tokenRepository,
+            final CreditBureauConfigurationRepositoryWrapper configDataRepository,
+            final CreditBureauTokenCommandFromApiJsonDeserializer fromApiJsonDeserializer) {
+        this.context = context;
+        this.tokenRepository = tokenRepository;
+        this.configDataRepository = configDataRepository;
+        this.fromApiJsonHelper = fromApiJsonHelper;
+        this.fromApiJsonDeserializer = fromApiJsonDeserializer;
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(ThitsaWorksCreditBureauIntegrationWritePlatformServiceImpl.class);
+
+    @Transactional
+    @Override
+    public String httpConnectionMethod(String process, String nrcID, String userName, String password, String subscriptionKey,
+            String subscriptionId, String url, String token, Long uniqueID, File report) {
+
+        String result = null;
+
+        try {
+            String post_params = null;
+            HttpURLConnection httpConnection = this.process(process, nrcID, userName, password, subscriptionKey, subscriptionId, url, token,
+                    uniqueID, report);
+
+            if (process.equals("token")) {
+                post_params = "" + "BODY=x-www-form-urlencoded&\r" + "grant_type=password&\r" + "userName=" + userName + "&\r" + "password="
+                        + password + "&\r";
+            } else if (process.equals("NRC")) {
+
+                post_params = "BODY=x-www-form-urlencoded&nrc=" + nrcID + "&";
+            } else if (process.equals("UploadCreditReport")) {
+
+                post_params = "BODY=formdata&" + report + "&" + "userName=" + userName + "&";
+            }
+
+            // token header not used when creating token i.e. when token will be null
+            if (token != null) {
+                httpConnection.setRequestProperty("Authorization", "Bearer " + token);
+            }
+
+            // this set is required only for (POST METHOD)- fetching uniqueID from NRC/Creating Token/Add Credit report
+            if (process.equals("NRC") || process.equals("token") || process.equals("UploadCreditReport")) {
+                httpConnection.setDoOutput(true);
+                OutputStream os = httpConnection.getOutputStream();
+                os.write(post_params.getBytes(StandardCharsets.UTF_8));
+                os.flush();
+                os.close();
+            }
+
+            result = this.httpResponse(httpConnection);
+
+        } catch (IOException e) {
+            LOG.error("Error occured.", e);
+        }
+        return result;
+
+    }
+
+    private HttpURLConnection process(String process, String nrcID, String userName, String password, String subscriptionKey,
```

Comment:
I think we can break this method down to two methods, getHTTPPostConnection() and getHTTPGetConnection(). Let me know if you agree.

Reply:
It would be even better not to code at this low level, but use a higher level ready made HTTP Client library. Do you know if https://square.github.io/okhttp/ and https://square.github.io/retrofit/? Both are already used in Fineract, search the existing code for examples how to use it...

Reply:
@rrpawar96 I see that you've switched to using OkHTTP now, looks great, thank you; I'm therefore resolving this comment. FYI @nikpawar89.